price-tolerance/src/main.nr

```diff
@@ -0,0 +1,34 @@
+mod float;
+use float::{ Float, lte, div, sub, sum, abs, printlnF };
+
+global MAX_VALIDATORS = 8;
+
+fn main(
+	validators: pub [Float; MAX_VALIDATORS],
+	tolerance: pub Float,
+) {
+	// Helpers
+	let n = validators.fold(0, |acc: u32, curr: Float| {
+		if curr.0 == 0 {
+			acc
+		}
+		else {
+			acc + 1
+		}
+	});
+	let values = validators.sort_via(lte).map(|x: Float| if x.0 == 0 { (1,0) } else { x });
+
+	// Compute median
+	let median = values[(MAX_VALIDATORS - n) + n / 2];
+	printlnF(median);
+
+	// Compute reports
+	for i in 0..MAX_VALIDATORS {
+		if values[i].0 != 0 {
+			let perc_diff = div(abs(sub(values[i], median)), div(sum(values[i], median), (2,0)));
```

Comment:
Percent change is defined as $$\frac{v_2 - v_1}{|v_2|}$$ while the percent difference is defined as $$\frac{|v_2 - v_1|}{\left(\frac{v_2+v_1}{2}\right)}$$.

The initial resource I used was related to economics: change is about how the same value is modified over some event (time, place, ...), while difference is about comparing two values in distinct groups (two different companies, sport teams, ...).

Reading more into it, it does indeed make more sense to use change, since our median will be reported as the "true" value, and we only care how far apart are validators from the truth. Also thanks @stefan-nikolov96